### PR TITLE
19714-ff-put-back-on

### PR DESCRIPTION
**Pull request by jamespaologarcia**
*Issue #:* /bcgov/entity#19714

Added FF put back on

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-621-bf6fiz2u.web.app

File: package.json
Comment (severinbeauvais) [inline package.json:3]: Please rebase so that this shows 7.0.42, and then maybe re-test to ensure that the other changes don't affect your updates.

File: src/mixins/allowable-actions-mixin.ts
Comment (severinbeauvais) [inline src/mixins/allowable-actions-mixin.ts:131]: As this line is a continuation of the previous line, please indent it.

While you're here, also please indent lines 121 and 126. Thanks!

Comment (severinbeauvais) [inline src/mixins/allowable-actions-mixin.ts:131]: When do you get if `GetFeatureFlag('supported-put-back-on-entities')` is falsey? I'm wondering if the `!!` at the beginning of the line changes line 131 to True? (which would be incorrect, since we don't want to display a feature that we can't be sure is enabled)

Comment (jamespaologarcia) [reply src/mixins/allowable-actions-mixin.ts:131]: I was just going for consistency because the code that checks feature flags are already using !!

Comment (jamespaologarcia) [reply src/mixins/allowable-actions-mixin.ts:131]: The way I see it though, it seems like a small price(1 character diff) for odd "just-in-case" scenarios

Comment (severinbeauvais) [reply src/mixins/allowable-actions-mixin.ts:131]: I'm not blaming you. I'm just having a critical look at the code. And yes, consistency is very good (even if it's wrong, because then everything is wrong and can be fixed together instead of separately).

But actually, I'm looking at this again and I think I must have had crossed eyes for a moment...

If GetFeatureFlag is null/undefined then the inner expression is undefined, but we do a `!!` on it (not a single !), so that will evaluate to False, which is what we want.

It's good as is. Carry on.

File: tests/unit/StaffNotation.spec.ts
Comment (severinbeauvais) [inline tests/unit/StaffNotation.spec.ts:67]: Since you're only testing Put Back On in this test suite (describe block), please delete the allowed filing types that you don't care about.

File: package.json
Comment (severinbeauvais) [reply package.json]: OK, you're rebased. Now please update the app version again 😛

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-621-bf6fiz2u.web.app

File: tests/unit/StaffNotation.spec.ts
Comment (severinbeauvais) [inline tests/unit/StaffNotation.spec.ts:300]: Did you mean to restore this, too?

File: tests/unit/StaffNotation.spec.ts
Comment (severinbeauvais) [inline tests/unit/StaffNotation.spec.ts:488]: Typo on line

**Review by severinbeauvais — COMMENTED** [review]
LGTM. I think it will OK after your next commit.

File: tests/unit/StaffNotation.spec.ts
Comment (jamespaologarcia) [reply tests/unit/StaffNotation.spec.ts]: For this one, I updated the unit test I've made specifically for this feature flag so that it would check if enabled/disabled on the menu items. It seemed redundant to still have this

Comment (jamespaologarcia) [reply tests/unit/StaffNotation.spec.ts]: image

Comment (severinbeauvais) [reply tests/unit/StaffNotation.spec.ts]: In this case, I don't mind some redundancy for the sake of consistency 😃

Comment (severinbeauvais) [reply tests/unit/StaffNotation.spec.ts]: I see what you mean about the double find in your screnshot above. Feel free to update it as you see fit.

File: tests/unit/StaffNotation.spec.ts
Comment (ketaki-deodhar) [inline tests/unit/StaffNotation.spec.ts:199]: any specific reason these two were swapped? Order matters in anyway?

Comment (jamespaologarcia) [reply tests/unit/StaffNotation.spec.ts:199]: No reason. I originally removed it to make the test case work and then found another solution so I put it back.

Comment (ketaki-deodhar) [reply tests/unit/StaffNotation.spec.ts:199]: No biggie but good to avoid unnecessary changes (not adding much value) :)
If possible, restore the way it was

Comment (jamespaologarcia) [reply tests/unit/StaffNotation.spec.ts]: Done

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-621-bf6fiz2u.web.app

**Review by ketaki-deodhar — APPROVED** [review]
LGTM! Please get for another approval before merging :)

File: tests/unit/StaffNotation.spec.ts
Comment (severinbeauvais) [inline tests/unit/StaffNotation.spec.ts:89]: Please stop removing whitespace in code. It's usually there to visually separate one code block from another. Also on line 182 below.

**Review by severinbeauvais — APPROVED** [review]

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-621-bf6fiz2u.web.app

**Pull request merged**
